os/include/tinyara/net/if/ble.h

```diff
@@ -0,0 +1,340 @@
+typedef trble_result_e (*trble_stop_scan)(struct netdev *dev);
+typedef trble_result_e (*trble_connect)(struct netdev *dev, trble_bd_addr *addr);
+typedef trble_result_e (*trble_disconnect_all)(struct netdev *dev);
+typedef trble_result_e (*trble_connected_device_list)(struct netdev *dev, trble_connected_list *out_connected_list);
+typedef trble_result_e (*trble_connected_info)(struct netdev *dev, trble_conn_handle conn_handle, trble_device_connected *out_connected_device);
+typedef trble_result_e (*trble_operation_enable_notification)(struct netdev *dev, trble_operation_handle *handle);
+typedef trble_result_e (*trble_operation_read)(struct netdev *dev, trble_operation_handle *handle, trble_data *out_data);
+typedef trble_result_e (*trble_operation_write)(struct netdev *dev, trble_operation_handle *handle, trble_data *in_data);
+typedef trble_result_e (*trble_operation_write_no_response)(struct netdev *dev, trble_operation_handle *handle, trble_data *in_data);
+
+/*** Peripheral(Server) ***/
+typedef trble_result_e (*trble_get_profile_count)(struct netdev *dev, uint16_t *count);
+// API for sending a characteristic value notification to the selected target(s). (notify to all clients conn_handle (notify all = 0x99))
+typedef trble_result_e (*trble_charact_notify)(struct netdev *dev, trble_attr_handle attr_handle, trble_conn_handle con_handle, trble_data *data);
+typedef trble_result_e (*trble_attr_set_data)(struct netdev *dev, trble_attr_handle attr_handle, trble_data *data);
+typedef trble_result_e (*trble_attr_get_data)(struct netdev *dev, trble_attr_handle attr_handle, trble_data *data);
+// reject attribute request in callback function and return error code
+typedef trble_result_e (*trble_attr_reject)(struct netdev *dev, trble_attr_handle attr_handle, uint8_t app_errorcode);
+typedef trble_result_e (*trble_get_mac_addr_by_conn_handle)(struct netdev *dev, trble_conn_handle con_handle, uint8_t bd_addr[TRBLE_BD_ADDR_MAX_LEN]);
+typedef trble_result_e (*trble_get_conn_handle_by_addr)(struct netdev *dev, uint8_t bd_addr[TRBLE_BD_ADDR_MAX_LEN], trble_conn_handle *con_handle);
+typedef trble_result_e (*trble_set_adv_data)(struct netdev *dev, trble_data *data);
+typedef trble_result_e (*trble_set_adv_resp)(struct netdev *dev, trble_data *data);
+typedef trble_result_e (*trble_conn_is_active)(struct netdev *dev, trble_conn_handle con_handle, bool *is_active);
+typedef trble_result_e (*trble_conn_is_any_active)(struct netdev *dev, bool *is_active);
+typedef trble_result_e (*trble_get_bonded_device)(struct netdev *dev, trble_bonded_device_list_s *device_list, uint16_t *device_count);
+typedef trble_result_e (*trble_start_adv)(struct netdev *dev);
+typedef trble_result_e (*trble_start_adv_directed)(struct netdev *dev, uint8_t bd_addr[TRBLE_BD_ADDR_MAX_LEN]);
+typedef trble_result_e (*trble_stop_adv)(struct netdev *dev);
+typedef trble_result_e (*trble_set_adv_interval)(struct netdev *dev, uint16_t interval);
```

Comment:
These function prototype declarations are different from those in da_ble_server.h, da_ble_client.h. Is this ok?

Reply:
@Pangwawaa 
You can ignore 'struct netdev *dev' param.  "trble_" will be called by BLE manager in user space.
In "trble_" functions, what these functions do is just call "rtw_ble_" functions.